spec/controllers/bookmarks_controller_spec.rb

```diff
@@ -252,12 +252,46 @@ def it_redirects_to_user_login
     let(:bookmark) { create(:bookmark, bookmarkable_id: chaptered_work.id) }
 
     context "when logged in" do
-      it "returns a bookmark on a public multi-chapter work" do
+      before do
         fake_login_known_user(bookmark.pseud.user)
+      end
+
+      it "returns a bookmark on a public multi-chapter work" do
         get :show, params: { id: bookmark }
         expect(response).to have_http_status(:success)
         expect(assigns(:bookmark)).to eq(bookmark)
       end
+
+      it "finds a work from a bookmark" do
+        get :show, params: { id: bookmark }
+        expect(response).to have_http_status(:success)
+        expect(assigns(:bookmark)).to eq(bookmark)
+        expect(assigns(:bookmark).bookmarkable).to eq(chaptered_work)
+      end
+    end
+  end
+
+  describe "fetch recent" do
+    # need several bookmarks on the same work
+    let(:work) { create(:work) }
+    let(:bookmark1) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "Extra 1") }
+    let!(:bookmark2) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "Different") }
+    let!(:bookmark3) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "I think this is great") }
+
+    # wouldn't normally need to render views in a controller test, but it's the only way to get the js response
```

Comment:
We could actually leave it out -- the JavaScript functionality is already covered by [a feature test](https://github.com/otwcode/otwarchive/blob/b63fce67a3975ec0614bb535c75d76c8dafbda71/features/bookmarks/bookmark_create.feature#L389-L411). It's just the [non-JavaScript lines this adds new coverage for](https://app.codecov.io/gh/otwcode/otwarchive/compare/3890/changes#D1L305).